tests/test_async_run_and_parse_tests.py

```diff
@@ -0,0 +1,1039 @@
+            f.write(mixed_module_code)
+            
+        with test_path.open("w") as f:
+            f.write(test_code)
+
+        tests_root = (Path(__file__).parent.resolve() / "../code_to_optimize/tests/pytest/").resolve()
+        project_root_path = (Path(__file__).parent / "..").resolve()
+
+        async_func = FunctionToOptimize(function_name="async_merge_sort", parents=[], file_path=Path(mixed_fto_path), is_async=True)
+
+        source_success, instrumented_source = instrument_source_module_with_async_decorators(
+            mixed_fto_path, async_func, TestingMode.BEHAVIOR
+        )
+
+        assert source_success
+        assert instrumented_source is not None
+        assert "@codeflash_behavior_async" in instrumented_source
+        assert "async def async_merge_sort" in instrumented_source
+        assert "def sync_quick_sort" in instrumented_source  # Should preserve sync function
+
+        mixed_fto_path.write_text(instrumented_source, "utf-8")
+        instrument_codeflash_capture(async_func, {}, tests_root)
+
+        opt = Optimizer(
+            Namespace(
+                project_root=project_root_path,
+                disable_telemetry=True,
+                tests_root=tests_root,
+                test_framework="pytest",
+                pytest_cmd="pytest",
+                experiment_id=None,
+                test_project_root=project_root_path,
+            )
+        )
+
+        test_env = os.environ.copy()
+        test_env["CODEFLASH_TEST_ITERATION"] = "0"
+        test_env["CODEFLASH_LOOP_INDEX"] = "1"
+        test_env["CODEFLASH_TEST_MODULE"] = "code_to_optimize.tests.pytest.test_mixed_sort_temp"
+        test_env["CODEFLASH_TEST_CLASS"] = ""
+        test_env["CODEFLASH_TEST_FUNCTION"] = "test_mixed_sorting"
+        test_env["CODEFLASH_CURRENT_LINE_ID"] = "0"
+        test_type = TestType.EXISTING_UNIT_TEST
+
+        func_optimizer = opt.create_function_optimizer(async_func)
+        func_optimizer.test_files = TestFiles(
+            test_files=[
+                TestFile(
+                    instrumented_behavior_file_path=test_path,
+                    test_type=test_type,
+                    original_file_path=test_path,
+                    benchmarking_file_path=test_path_perf,
+                )
+            ]
+        )
+
+        test_results, _ = func_optimizer.run_and_parse_tests(
+            testing_type=TestingMode.BEHAVIOR,
+            test_env=test_env,
+            test_files=func_optimizer.test_files,
+            optimization_iteration=0,
+            pytest_min_loops=1,
+            pytest_max_loops=1,
+            testing_time=0.1,
+        )
+
+        assert test_results is not None
+        assert test_results.test_results is not None
+        
+        results_list = test_results.test_results
+        async_calls = [r for r in results_list if r.id.function_getting_tested == "async_merge_sort"]
+        assert len(async_calls) >= 1
```

Comment:
this test needs to be improved. See all the properties we are testing in the sync version of this, and test those as well.
Things like - the return values, the test id.

Reply:
This is important to give us confidence that these critical parameters are correct and never broken in the future.